semgrep-core/src/parsing/Test_parsing.ml

```diff
@@ -186,12 +186,52 @@ let test_parse_tree_sitter lang xs =
 *)
 let parsing_common ?(verbose = true) lang xs =
   let timeout_seconds = 10.0 in
+  (* Without the use of Memory_limit below, we were getting some
+   * 'Fatal error: out of memory' errors in the parsing stat CI job,
+   * because of some ulimit -v 4000000 in the run-lang script,
+   * which then was aborting the whole job.
+   *
+   * Note that the mem_limit_mb value below must be less than the
+   * ulimit value in run-lang.
+   *
+   * In fact, it is not clear how much less it must be. For example with:
+   * $ ulimit -v 1000000
+   * $ semgrep-core -lang c -parsing_stats -json lang/c/tmp/php-php-src/ext/fileinfo/
+   * I still get a core dump with a "Fatal error: out of memory",
+   * even if I set mem_limit_mb below to 500.
+   * Seems like the Gc alarm does not trigger, maybe because there is no
+   * major GC cycle triggered before reaching the 1 GB ulimit.
+   *
+   * So the current workaround is to still put a limit below, but
```

Comment:
> ... put a limit below, ...

I am not sure what is the reference for "below" here. I mean, below what exactly? Before it was below the `ulimit -v` but now that does not exist. Below the memory available in the system?